pinot-query-planner/src/main/java/org/apache/pinot/calcite/sql/fun/PinotOperatorTable.java

```diff
@@ -69,6 +72,30 @@ public static PinotOperatorTable instance() {
     return INSTANCE.get();
   }
 
+  // The standard Calcite + and - operators don't support operations on TIMESTAMP types. However, Pinot supports these
+  // operations, so we need to define our own operators. Note that Postgres supports - on TIMESTAMP types, but not +.
+  // Calcite only supports such operations if the second operand is an interval (similar to Postgres for the +
+  // operator).
+  public static final SqlBinaryOperator PINOT_PLUS =
```

Comment:
Do we need to explicitly register them? Or just let the scalar function register it which has very loose check?

Reply:
That works for `ADD` / `SUB`, but not for `+` / `-` since those are unique operators. We get an error like this during query compilation (sample query - `SELECT ts - CAST(1234567890 AS TIMESTAMP) from airlineStats LIMIT 1;`) if we try to register `+` / `-` as Pinot scalar functions - 

```
Invalid number of arguments to function '-'. Was expecting 2 arguments
```

Reply:
I think that should be fixable by adding a new constructor to `PinotSqlFunction` and allowing caller to pass in `SqlKind` though.